internal/authorization/authorizer.go

```diff
@@ -47,8 +47,16 @@ func selectMatchingSubjectRules(rules []schema.ACLRule, subject Subject) []schem
 	selectedRules := []schema.ACLRule{}
 
 	for _, rule := range rules {
-		if isSubjectMatching(subject, rule.Subject) && isIPMatching(subject.IP, rule.Networks) {
-			selectedRules = append(selectedRules, rule)
+		if len(rule.Subjects) > 0 {
+			for _, subjectRule := range rule.Subjects {
+				if isSubjectMatching(subject, subjectRule) && isIPMatching(subject.IP, rule.Networks) {
+					selectedRules = append(selectedRules, rule)
+				}
+			}
+		} else {
+			if isSubjectMatching(subject, "") && isIPMatching(subject.IP, rule.Networks) {
```

Comment:
Why using `isSubjectMatching(subject, "") && ` in the condition? From what I can see, this will always return true according to the implementation: https://github.com/authelia/authelia/blob/4d1c2f38bef9708463b3f7ede4f12043de8df939/internal/authorization/subject_matcher.go#L11.
So IMO, we can just strip it.

Reply:
You're right. I originally had left this because I wanted to preserve how it was being processed before without looking at the implementation details. But it always returns true in this scenario so I just stripped it out.